NgnetMan/src/components/Request.js

```diff
@@ -2,68 +2,51 @@ import { useContext, useState } from 'react'
 import Select from 'react-select'
 import { HttpContext } from '../contexts/httpContext';
 import * as authService from '../services/authService'
+import common from '../common/common.json'
 
 export function Request() {
 
-    const actions = [
-        { value: 'register', label: 'Register' },
-        { value: 'login', label: 'Login' },
-        { value: 'profile', label: 'Profile' },
-        { value: 'logout', label: 'Logout' }
-    ];
+    const actions = common.actions;
+    const bodyParams = common.params;
 
-    let [actionState, setActionState] = useState(actions);
-    let [paramState, setParamState] = useState();
+    let [action, setAction] = useState();
     let { setResponse } = useContext(HttpContext);
 
-    const bodyParams = {
-        register: [
-            { value: 'email', label: 'Email' },
-            { value: 'username', label: 'Username' },
-            { value: 'password', label: 'Password' },
-            { value: 'repeatPassword', label: 'RepeatPassword' }
-        ],
-        login: [
-            { value: 'username', label: 'Username' },
-            { value: 'password', label: 'Password' }
-        ],
-    };
-
     function sendRequest(e) {
         e.preventDefault();
         let formData = new FormData(e.currentTarget);
         let data;
 
-        if (bodyParams[paramState]) {
-            data = bodyParams[paramState].reduce((acc, curr) => {
+        if (common.params[action]) {
+            data = common.params[action].reduce((acc, curr) => {
                 let key = curr.value;
                 acc[key] = formData.get(curr.value);
```

Comment:
Add code below instead of row: 23 to replace empty strings with nulls
const value = formData.get(curr.value);
acc[key] = value ? value : null;